Classes/Service/DemandService.php

```diff
@@ -187,6 +231,91 @@ public function getInnerBoundsForProperty(string $propertyName): array
      */
     protected function getConfiguredDemandProviders(): array
     {
-        // TODO: Initialize DemandProvider objects configured in TypoScript and return in order of execution
+        $configurationManager = GeneralUtility::makeInstance(ConfigurationManager::class);
+        $config = $configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_FULL_TYPOSCRIPT)['config.']['tx_demander.'];
+        $demandProviders = [];
+
+        foreach ($config['demandProviders.'] as $id => $demandProvider)
+        {
+            $demandProviders[$id] = GeneralUtility::makeInstance($demandProvider);
+        }
+        return $demandProviders;
+    }
+
+    /**
+     * Transform necessary values from string to integer.
+     *
+     * @param array $restrictionsArray
+     * @return array
+     */
+    protected function restrictionsToInt(array $restrictionsArray): array
+    {
+        $restrictions = [];
+
+        foreach ($restrictionsArray as $key => $restriction) {
+            if ($restriction['value']) {
+                $value = (is_numeric($restriction['value'])) ? (int)$restriction['value'] : $restriction['value'];
+                $restriction = array_replace($restriction, ['value' => $value]);
+                $restrictions[$key] = $restriction;
+            }else{
+                $restrictions[$key] = $this->restrictionsToInt($restriction);
+            }
+        }
+
+        return $restrictions;
+    }
+
+    /**
+     * Resolve field name from alias or from array if OR, AND provided.
+     *
+     * @param mixed $alias
+     * @param null|string $rootKey
+     * @return mixed
+     */
+    protected function getFieldPropsFromAlias($alias, $rootKey = null)
```

Comment:
1. I think you mean "properties", but write "props". Don't use abbreviations and spell out the full name.

2. Missing return type.

```suggestion
    protected function getFieldPropertiesFromAlias($alias, $rootKey = null): ?string
```

Reply:
It must returns string or array, but in your example it returns string or null, how could I declare that return type?

Reply:
Right. Settle on one of the return types. That way you avoid having to handle different return types on the other end. I guess array is the best, so I suggest you do that. 

Maybe you can even avoid a null value, just return empty array?